server/controllers/documentController.js

```diff
@@ -2,181 +2,196 @@ import db from '../models';
-      })
-      .then((document) => {
-        res.status(201).send({
-          message: 'Document created successfully',
-          data: document
-        });
-      })
-      .catch((err) => {
-        res.status(400).send({
-          message: 'There was an error while creating the document', err
-        });
+  createNewDocument(request, response) {
+    if (!request.body.title || !request.body.content) {
+      return response.status(400).send({
+        message:
+        'The title or the content of the document is empty'
+      });
+    }
+    db.documents.findOne({ where: { title: request.body.title } })
+      .then((documentExists) => {
+        if (documentExists) {
+          return response.status(400).send({
+            message: 'This document already exists'
+          });
+        }
+        db.documents
+          .create({
+            title: request.body.title,
+            content: request.body.content,
+            access: request.body.access || 'public',
+            userId: request.decodedToken.userId,
+            userRoleId: request.decodedToken.roleId
+          })
+          .then((document) => {
+            response.status(201).send({
+              message: 'Document created successfully',
+              data: document
+            });
+          })
+          .catch((err) => {
+            response.status(400).send({
+              message: 'There was an error while creating the document', err
+            });
+          });
       });
   },
 
-  updateDocument(req, res) {
+  updateDocument(request, response) {
+    if (!request.body.title && !request.body.content) {
+      return response.status(400).send({
+        message: 'No update detected'
+      });
+    }
     db.documents
-      .findById(req.params.id)
+      .findById(request.params.id)
       .then((doc) => {
         if (!doc) {
-          return res.status(404).send({
+          return response.status(404).send({
             message:
-            'The document was not found'
+            'Cannot update a document that does not exist'
           });
         }
-        if (parseInt(doc.userId, 10) === req.decodedToken.userId) {
+        if (parseInt(doc.userId, 10) === request.decodedToken.userId) {
           doc.update({
-            title: req.body.title || doc.title,
-            content: req.body.content || doc.content,
-            access: req.body.access || doc.access
+            title: request.body.title || doc.title,
+            content: request.body.content || doc.content,
+            access: request.body.access || doc.access
           })
-            .then(updatedDoc => res.status(200).send({
+            .then(updatedDoc => response.status(200).send({
               message: 'The document was updated successfully',
               data: updatedDoc
             }));
         } else {
-          res.status(401).send({ message: 'Permission denied' });
+          response.status(401).send({ message: 'Permission denied' });
```

Comment:
The 401 status code is used when a user is not authorized to access a particular resource, this could be (s)he is not logged in. The appropriate status code for this response is 403(forbidden) since the user is logged in but does not have the right privilege to access that resource. You should take a look at [http status code](http://www.restapitutorial.com/httpstatuscodes.html)